src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -0,0 +1,1292 @@
+
+            if (isInline)
+            {
+                this.WriteInlineExpression(value.Value);
+            }
+        }
+
+        internal override void WriteRecordExpressionElementHeader(IEdmRecordExpression expression)
+        {
+            // Record expressions are represented as objects with one member per property value expression.
+            this.jsonWriter.WriteStartObject();
+
+            if (expression.DeclaredType != null)
+            {
+                this.jsonWriter.WriteRequiredProperty("$Type", expression.DeclaredType.FullName());
+            }
+            // It MAY contain annotations for itself. It's not supported now.
+        }
+
+        internal override void WritePropertyConstructorElementHeader(IEdmPropertyConstructor constructor, bool isInline)
+        {
+            // The member name is the property name, and the member value is the property value expression.
+            this.jsonWriter.WritePropertyName(constructor.Name);
+
+            if (isInline)
+            {
+                this.WriteInlineExpression(constructor.Value);
+            }
+
+            // Annotations for record members are prefixed with the member name. It's not supported now.
+        }
+
+        internal override void WritePropertyConstructorElementEnd(IEdmPropertyConstructor constructor)
+        {
+            // nothing here.
+        }
+
+        internal override void WriteStringConstantExpressionElement(IEdmStringConstantExpression expression)
+        {
+            // String expressions are represented as a JSON string.
+            this.jsonWriter.WriteStringValue(expression.Value);
+        }
+
+        internal override void WriteBinaryConstantExpressionElement(IEdmBinaryConstantExpression expression)
+        {
+            // Binary expressions are represented as a string containing the base64url-encoded binary value.
+            this.jsonWriter.WriteStringValue(BinaryToString(expression));
+        }
+
+        internal override void WriteBooleanConstantExpressionElement(IEdmBooleanConstantExpression expression)
+        {
+            // Boolean expressions are represented as the literals true or false.
+            this.jsonWriter.WriteBooleanValue(expression.Value);
+        }
+
+        internal override void WriteNullConstantExpressionElement(IEdmNullExpression expression)
+        {
+            // Null expressions that do not contain annotations are represented as the literal null.
+            this.jsonWriter.WriteNullValue();
+
+            // Null expression containing annotations are represented as an object with a member $Null whose value is the literal null.
+            // So far, it's not supported.
+        }
+
+        internal override void WriteDateConstantExpressionElement(IEdmDateConstantExpression expression)
+        {
+            // Date expressions are represented as a string containing the date value.
+            this.jsonWriter.WriteStringValue(expression.Value.ToString());

```

Comment:
>.ToString [](start = 61, length = 9)

why not "EdmValueWriter.DateAsXml(expression.Value)" (for consistency with the other date/time values, and with XML)?  Are we sure that ToString() for a date expression will always return the right format?

Reply:
1) I try to get rid of XML in json writer.

2) I saw there's a static method as "DateAsXml":
```C#   
    internal static string DateAsXml(Date d)
        {
            var value = d.ToString();
            return value;
        }
```
From this, it calls "ToString()" directly.

So, from #1, #2, I think "Why don't we call "ToString" directly. 


---
In reply to: [443768300](https://github.com/OData/odata.net/pull/1775#discussion_r443768300) [](ancestors = 443768300)